smithy-aws-endpoints/src/main/java/software/amazon/smithy/rulesengine/aws/validators/EndpointModifierValidator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.rulesengine.aws.validators;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import software.amazon.smithy.model.FromSourceLocation;
+import software.amazon.smithy.model.Model;
+import software.amazon.smithy.model.shapes.ServiceShape;
+import software.amazon.smithy.model.validation.AbstractValidator;
+import software.amazon.smithy.model.validation.ValidationEvent;
+import software.amazon.smithy.rulesengine.aws.traits.StandardPartitionalEndpointsTrait;
+import software.amazon.smithy.rulesengine.aws.traits.StandardRegionalEndpointsTrait;
+import software.amazon.smithy.utils.SetUtils;
+
+/**
+ * Validate EndpointModifier traits that are applied on a service.
+ */
+public class EndpointModifierValidator extends AbstractValidator {
```

Comment:
Should be `final` and I think a better name for this would be `AwsStandardEndpointPatternValidator`.